### DISCUSSION TITLE
Why pop-launcher is not handling the desktop entries activation directly

### DISCUSSION THREAD

**Discussion by oknozor** [pop-os/launcher#118] [General]
Currently the launcher activate plugins entries other than desktop entries directly (running terminal, xdg-open etc). 

So I was wondering why is this different for desktop entries ? Shouldn't the launcher take care of the exec part instead of returning the path to de entry file on activation ? 

I suspect there is a good reason for this but I was not able to figure it out.

**Comment by mmstick** [comment]
I had issues getting focus of windows launched by the pop-launcher process instead of through GNOME Shell's own desktop launch functions. I'm unsure if that's still an issue.

**Comment by soloturn** [comment]
sorry for the beginner question, but what is pop-launcher doing, and what cosmic-launcher is doing, or, in other words, what happens if such launchers would not exist, @mmstick ?

**Reply by mmstick** [reply]
pop-launcher is a launcher backend. It uses JSON to communicate with a GUI frontend. The cosmic-launcher is a GUI COSMIC applet frontend for pop-launcher. If neither existed, there would be no launcher in Pop!_OS's pop-shell, or in COSMIC.
